vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Space.java

```diff
@@ -167,5 +194,18 @@ public Boolean hasIncomingLinks() {
     public void setIncomingLinks(Boolean incomingLinks) {
         this.incomingLinks = incomingLinks;
     }
-    
+
+    @Override
+    public boolean isHideIncomingLinks() {
+        return hideIncomingLinks;
+    }
+
+    @Override
+    public void setHideIncomingLinks(Boolean hideIncomingLinks) {
+        if (hideIncomingLinks == null) {
+            this.hideIncomingLinks = false;
+        }
+        this.hideIncomingLinks = hideIncomingLinks;
```

Comment:
if `hideInComingLinks` is null, this line will set `this.hideInComingLinks` to null.

Reply:
It's throwing an error, since the variable is boolean.
Shall I cast the variable as Eclipse is suggesting?
`this.hideIncomingLinks = (Boolean) null;`

Reply:
what would that accomplish? and you also would still have the same problem. You set `hideIncomingLinks` to false and then you set it back to null.

Reply:
But when it's null, I will have to set to a default value `false`, so that it can check further in `ExhibitionSpaceController` file whether it's `true/false` right?

Reply:
Or should I set the type of `hideIncomingLinks` as primitive `boolean`?

Reply:
> But when it's null, I will have to set to a default value `false`, so that it can check further in `ExhibitionSpaceController` file whether it's `true/false` right?

Please go line by line and check what is happening in every line. It's not what you think. ;)

Reply:
what happened to the case that `hideIncomingLinks` is null. Can this not happen anymore?

Reply:
No, it shouldn't happen. Since `hideIncomingLinks` is a `boolean` primitive type by default it's false. So it should be fine.

Reply:
Have you tried what happens if the value is `null` in the db (for spaces that existed before this property was added)? Hibernate would probably use the setter method to set  `hideIncomingLinks` and pass `null`, which would most likely create an exception is my guess.

Reply:
I currently tested these changes, on the existing Spaces. It seems to work fine. 
Or if I use the previous code, then the exception is getting handled when using `Boolean` type and the setter method sets `hideIncomingLinks` to `false` when it is `null`.